text/0000-rust-spec.md

```diff
@@ -0,0 +1,171 @@
+the [source code](https://github.com/rust-lang/rust/),
+and more.
+
+These are currently all incomplete, and/or not a good source to rely on.
+
+More background information is available in [this blog post](https://blog.m-ou.se/rust-standard/).
+
+# Goals and non-goals
+
+- The goal of the Rust specification work is the creation of a document, the Rust specification.
+
+- The goal is _not_ to change how the language evolves;
+  the relevant teams (Language, Libs-API, …) remain in charge of the evolution of their respective parts of Rust,
+  and will continue to use processes as they see fit (e.g. RFCs).
+
+- The specification will only be considered "official" once the relevant teams have approved of its contents.
+  Changes to the official specification must be approved by the relevant team(s).
+
+- The goal is to serve the needs of Rust users, such as
+  authors of unsafe Rust code, those working on safety critical Rust software,
+  language designers, maintainers of Rust tooling, and so on.
+
+- It is _not_ a primary goal of the specification to aid in the development of alternative Rust implementations,
+  although authors of alternative compilers might still find the specification to be useful.
+
+  What this means is that, unlike the C or C++ standard,
+  the Rust specification does not provide a set of requirements for a compiler to be able to call itself "a Rust™ compiler".
+  Instead, it specifies the behaviour of the Rust compiler.
+  (So, not "A Rust implementation should …", but instead "Rust will …".)
+
+- The scope remains to be determined, but at least includes all topics currently included in 
+  the (incomplete) [Rust Reference](https://doc.rust-lang.org/stable/reference/).
+
+- The Rust specification is expected to replace the current Rust Reference.
+
+- The scope of the specification can grow over time, depending on the needs of Rust users and
+  time and motivation of those working on it.
+
+  For example, it might grow over time to also specify details of Cargo,
+  compiler flags, procedural macros, or other parts that we might initially consider out of scope.
+
+- The specification is specific to the latest version of Rust.
+  A copy of the specification is included with each Rust release,
+  as we currently already do with much of our documentation.
+
+  While the specification might include notes about the Rust version that a feature was introduced in
+  for informative purposes (similar to standard library documentation),
+  it does not attempt to accurately specify older, unsupported versions of Rust.
+
+- The specification specifies all Rust _editions_, as supported by the latest version of the Rust compiler.
+
+- Once the specification reaches an initial usable version,
+  the relevant teams are expected to incorporate it in their process for language evolution.
+  For example, the language team could require a new language feature to be included
+  in the specification as a requirement for stabilization.
```

Comment:
At the risk of over-specifying things, who would/might be expected to write the language for the standard? My concern is that adding more steps, particularly nontrivial ones, could slow development a fair amount. While I could probably write a spec for some things that I propose, doing so takes _a lot_ of time. Is this where the later-mentioned technical editor would come in?

Reply:
At what point will it be required that a spec exists for new features? If it's never required, the spec becomes nom-normative and turns into the Reference we have today, but doing it in an RFC adds a burden like @jhpratt mentioned. My preference is that documenting features in the spec is required before they're stabilized.